crates/interfaces/src/db.rs

```diff
@@ -1,38 +1,39 @@
+use std::fmt::Display;
 use thiserror::Error;
 
 /// Database error type.
 #[derive(Clone, Debug, PartialEq, Eq, Error)]
 pub enum DatabaseError {
```

Comment:
I'd prefer if we don't box, so instead we'd remove the `#[error]` proc macro usage here and impl `Display` and `Error` manually, where we can parse the error code. Wdyt?

Reply:
I don't see how this is different from a macro implementation

Reply:
Error code is specific to implementation, so the idea is to introduce a new abstract method to parse `i32` into a string?